ui/src/app/registration/modal/modal.component.ts

```diff
@@ -6,6 +6,12 @@ import { RegisterUserRequest } from 'src/app/shared/jsonrpc/request/registerUser
 import { Service, Websocket } from 'src/app/shared/shared';
 import { environment } from 'src/environments';
 
+function emailValidator(control: FormControl): { [s: string]: boolean } | null {
+  const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/; // Regex to check for valid email format
```

Comment:
Validating an email with regex is unfortunately very complicated - see e.g. this thread on StackOverflow: https://stackoverflow.com/questions/201323/how-can-i-validate-an-email-address-using-a-regular-expression

After briefly reading through the thread I think your implementation is currently incomplete. Can we not reuse the `Validators.email` as before?